base_layer/core/src/transactions/transaction_components/encrypted_value.rs

```diff
@@ -32,31 +32,29 @@ use crate::consensus::{ConsensusDecoding, ConsensusEncoding};
 
 const SIZE: usize = 24;
 
-#[derive(Debug, Clone, Deserialize, Serialize)]
-pub struct EncryptedValue {
-    /// value: u64 + tag: [u8; 16]
-    pub data: [u8; SIZE],
-}
+/// value: u64 + tag: [u8; 16]
+#[derive(Debug, Clone, Deserialize, Serialize, PartialEq, Eq, Hash)]
+pub struct EncryptedValue(#[serde(with = "hex::serde")] pub [u8; SIZE]);
```

Comment:
Any reason we use hex to serialize this value?
This essentially doubles the storage required to 48 bytes.
You could implement the `Hex` trait from tari_utilities if you want to convert this value to and from hex

Reply:
I've added hex support to read values from the `dibbler_faucet.json` file. Ok, I'll try to make it more coherently with other types we use in the `TransactionOutput`.

Reply:
:+1: I think you may need serialize_with, something like this https://stackoverflow.com/a/67148611/1077579
Essentially, using hex only when the serialisation is "human readable"